module/Kanbanize/src/Kanbanize/Service/Importer.php

```diff
@@ -0,0 +1,442 @@
+	 * @return Stream
+	 */
+	private function updateStream($project, $board, ReadModelStream $s) {
+		$subject = $this->createFormattedSubject($project['name'], $board['name']);
+		$stream = $this->streamService->getStream($s->getId());
+		if($s->getSubject() != $subject ){
+			$this->transactionManager->beginTransaction();
+			try{
+				$stream->setSubject($subject, $this->requestedBy);
+				$this->transactionManager->commit();
+				$this->updatedStreams++;
+			}catch (\Exception $e) {
+				$this->transactionManager->rollback();
+				throw $e;
+			}
+		}
+		return $stream;
+	}
+	
+	/**
+	 * 
+	 * @param array $project
+	 * @param array $board
+	 * @throws Exception
+	 * @return Stream
+	 */
+	private function createStream($project, $board) {
+		$subject = $this->createFormattedSubject($project['name'], $board['name']);
+		$options = [
+			'boardId' => $board['id'],
+			'projectId' => $project['id']
+		];
+	
+		$this->transactionManager->beginTransaction();
+		try {
+			$stream = KanbanizeStream::create($this->organization, $subject, $this->requestedBy, $options);
+			$this->streamService->addStream($stream);
+			$this->transactionManager->commit();
+			$this->createdStreams++;
+			return $stream;
+		}catch (\Exception $e) {
+			$this->transactionManager->rollback();
+			throw $e;
+		}
+	}
+	/**
+	 * @param tasksFound
+	 * @param ex
+	 * @param kanbanizeTasks
+	 * @param readModelTask
+	 * @param taskId
+	 * @param task
+	 * @param createdTaskId
+	 */
+	public function importTasks($project, $board, Stream $stream) {
+		try{
+			$kanbanizeTasks = $this->api->getAllTasks($board['id']);
+			$tasksFound = [];
+			foreach($kanbanizeTasks as $kanbanizeTask){
+				try{
+					$task = $this->importTask ( $project, $board, $stream, $kanbanizeTask );
+					$tasksFound[] = $task->getId();
+				}catch (\Exception $e){
+					$this->errors[] = "Cannot import task {taskId: {$kanbanizeTask['taskid']}, boardId: {$board['id']}, projectId: {$project['id']}} due to {$e->getMessage()}";
+					$tasksFound[] = $kanbanizeTask['taskid'];
+				}
+			}
+			$this->deleteTasks($stream, $tasksFound);
+		}catch(KanbanizeApiException $e){
+			$this->errors[] = "Cannot import tasks due to {$e->getMessage()}";
+		}}
+	/**
+	 * 
+	 * @param string $project
+	 * @param string $board
+	 * @param Stream $stream
+	 * @param array $kanbanizeTask
+	 * @return \Kanbanize\KanbanizeTask|Task
+	 */
+	public function importTask($project, $board, Stream $stream, $kanbanizeTask) {
+		$readModelTask = $this->kanbanizeService->findByTaskId($kanbanizeTask['taskid']); //TODO: esplorare nuovi metadati per l'event store
+		if(is_null($readModelTask)){
+			return $this->createTask($kanbanizeTask, $stream);
+		}
+		$task = $this->taskService->getTask($readModelTask->getId());
+		return $this->updateTask($project, $board, $task, $kanbanizeTask);
+	}
+
+	private function updateTask($project, $board, Task $task, $kanbanizeTask){
+		$updatedTask = false;
+		$this->transactionManager->beginTransaction();
```

Comment:
Perchè, isolando in ogni transazione l'aggiornamento di un solo campo, se un aggiornamento dovesse fallire posso comunque tentare di aggiornare il resto